numbat/src/tokenizer.rs

```diff
@@ -171,8 +171,16 @@ fn is_other_allowed_identifier_char(c: char) -> bool {
     c == '%'
 }
 
+fn is_subscript_char(c: char) -> bool {
+    let c_u32 = c as u32;
+
+    // See https://en.wikipedia.org/wiki/Unicode_subscripts_and_superscripts#Superscripts_and_subscripts_block
+    (0x2080..=0x209CF).contains(&c_u32)
+}
+
 fn is_identifier_start(c: char) -> bool {
     unicode_ident::is_xid_start(c)
+        || is_subscript_char(c)
```

Comment:
Do we want to allow subscripts at the beginning of an identifier?

Reply:
Good question. I personally think subscripts at the start of identifiers look a bit unaesthetic, but technically speaking I see no reason not to allow them there.

Reply:
I think it's not a question of aesthetics, but rather about being future-proof. If we want to use subscript characters in the syntax at some point in the future, it might be better to only allow them as identifier-continue characters.

It's subtle and might never be required, but if we don't see a use case for identifiers that start with a subscript, I would only allow them as identifier-continue characters.